### ISSUE TITLE
Missing peer dependency `micromark-core-commonmark` when using PNPM

### ISSUE THREAD

**Issue by MrHaila** [DavidAnson/markdownlint-cli2#483]
After updating to `markdownlint-cli2` 0.17.0 I started getting an error in my project that uses PNPM:

```
node:internal/modules/esm/resolve:838
   throw new ERR_MODULE_NOT_FOUND(packageName, fileURLToPath(base), null);
         ^
 
 Error [ERR_MODULE_NOT_FOUND]: Cannot find package 'micromark-core-commonmark' imported from /REDACTED/node_modules/.pnpm/markdownlint@0.37.2/node_modules/markdownlint/lib/micromark-parse.mjs
     at packageResolve (node:internal/modules/esm/resolve:838:9)
     at moduleResolve (node:internal/modules/esm/resolve:907:18)
     at defaultResolve (node:internal/modules/esm/resolve:1037:11)
     at ModuleLoader.defaultResolve (node:internal/modules/esm/loader:650:12)
    at #cachedDefaultResolve (node:internal/modules/esm/loader:599:25)
     at ModuleLoader.resolve (node:internal/modules/esm/loader:582:38)
     at ModuleLoader.getModuleJobForImport (node:internal/modules/esm/loader:241:38)
     at ModuleJob._link (node:internal/modules/esm/module_job:132:49) {
   code: 'ERR_MODULE_NOT_FOUND'
 }
 
 Node.js v22.11.0
```

I believe this is because PNPM enforces isolation between dependencies and in this case maybe `markdownlint` or something else in the chain tries to depend on `micromark-core-commonmark` without actually having it listed in its (peer) dependencies.

My immediate workaround for this that may help others was to add `hoist-pattern[]=micromark-core-commonmark` into my `.npmrc` so the dependency in question became globally available for all other dependencies.

**Comment by DavidAnson** [comment]
There is a single, deliberate case of a dependency not being declared: https://github.com/DavidAnson/markdownlint/blob/a1081ec450f9f229d604a0a305ff4498557a336d/lib/micromark-parse.mjs#L9-L11

As the comment says, it is necessary for that import to produce **exactly** the same module that is explicitly imported by micromark. Otherwise, the shim that is used to capture unresolved reference definitions will not work and the rules that depend on it will not detect certain issues.

I will see if I can reproduce this behavior and whether setting it up as a peer dependency (perhaps with version "*") makes pnpm happy and doesn't break this scenario.

I don't know what effect your workaround will have, so for the time being I would recommend not updating CLI2.

**Comment by DavidAnson** [comment]
Alternatively, the tip at the bottom of this page suggests how to get pnpm to use a similar "hoisted" directory structure to npm (as a workaround that would allow you to upgrade now): https://pnpm.io/motivation

**Comment by DavidAnson** [comment]
This is verified fixed in `markdownlint@0.37.3` and should be fixed in `markdownlint-cli2@0.17.1` (though it doesn't reproduce in CI, so is unverified).

**Comment by DavidAnson** [comment]
I found a way to verify the fix works for `markdownlint-cli2` as well, this should work for you now.
